testsuite/shared/src/main/java/org/jboss/as/test/deployment/DeploymentInfoUtils.java

```diff
@@ -0,0 +1,797 @@
+     * Verify by management trusted command.
+     *
+     * @param param Containers of parameters, because loot of optional parameters
+     * @throws CommandFormatException Throw in case of assert failure
+     * @throws IOException            Throw in case of problem with execute management command
+     */
+    private static void verifyCheck(InternalParameters param) throws CommandFormatException, IOException {
+        if (param.getCtx() == null) {
+            log.warn("Skip double checking by management trusted commands - CommandContext connection not set!");
+            return;
+        }
+        if (param.isSearchTypeStatus() && UNKNOWN.equals(param.getExpectedState())) {
+            throw new IllegalStateException("Could not verify deployment state " + UNKNOWN + "!");
+        }
+
+        log.info("Double checking " + param.getName() + " with management command trusted for " + param.getGoalStr());
+
+        if (param.getCtx().isTerminated()) {
+            throw new IllegalStateException("FAILED: Could not double checking " + param.getName() +
+                    " with management trusted command for " + param.getGoalStr() + "!" + "Because connection to cli is closed!");
+        }
+
+        if (param.isSearchTypeStatus()) {
+            verifyCheckStatus(param);
+        } else if (param.isSearchTypeExist()) {
+            verifyCheckExist(param);
+        } else if (param.isSearchTypeMissing()) {
+            verifyCheckMissing(param);
+        } else {
+            throw new IllegalStateException("Unknown operation selected! Could not verify check!");
+        }
+
+        log.info("Double checking " + param.getName() + " with management trusted command for " + param.getGoalStr() + " - Success");
+    }
+
+    /**
+     * Concentration of verify status of application deployment.
+     * Double checking for state and existence of application deployment and verify list and Info result.
+     * Verify by management trusted command.
+     *
+     * @param param Containers of parameters, because loot of optional parameters
+     * @throws CommandFormatException Throw in case of assert failure
+     * @throws IOException            Throw in case of problem with execute management command
+     */
+    private static void verifyCheckStatus(InternalParameters param) throws CommandFormatException, IOException {
+        String serverGroupStr = param.getResult().hasServerGroup() ? "/server-group=" + param.getResult().getServerGroup() : "";
+        param.getResult().setRequest(serverGroupStr + "/deployment=" + param.getName() + ":read-attribute(name=enabled)");
+        ModelNode mn = param.getCtx().buildRequest(param.getResult().getRequest());
+        ModelNode response = param.getCtx().getModelControllerClient().execute(mn);
+        param.getResult().setResponse(response.asString());
+
+        // State NOT_ADDED is not supported by management command
+        if (!NOT_ADDED.equals(param.getExpectedState())) {
```

Comment:
!NOT_ADDED is not that easy to read, an advise would be to reverse your if branches, if(NOT_ADDED) {...} else {...}. In general, the simpler your conditions, the better.

Reply:
Ok. Thank you. Done